crates/interpreter/src/valid.rs

```diff
@@ -129,13 +129,24 @@ impl<'m> Context<'m> {
         let mut side_tables = vec![];
         if let Some(mut parser) = self.check_section(parser, SectionId::Code)? {
             check(self.funcs.len() == imported_funcs + parser.parse_vec()?)?;
+            let saved = parser.save();
             for x in imported_funcs .. self.funcs.len() {
                 let size = parser.parse_u32()? as usize;
                 let mut parser = parser.split_at(size)?;
                 let t = self.functype(x as FuncIdx).unwrap();
                 let mut locals = t.params.to_vec();
                 parser.parse_locals(&mut locals)?;
-                side_tables.push(Expr::check_body(self, &mut parser, &refs, locals, t.results)?);
+                let parser_start =
+                    unsafe { parser.save().as_ptr().offset_from(saved.as_ptr()) as usize };
```

Comment:
We don't need pointer arithmetic, we can just compute using `usize`, since we'll store the result to flash and need to use integers anyway.